flake.nix

```diff
@@ -1,8 +1,37 @@
 {
   description = "Poetry2nix flake";
 
-  outputs = { self }:
+  outputs = { self, nixpkgs }:
+    let
+      # TODO: There must be a better way to provide arch-agnostic flakes..
+      systems = [ "x86_64-linux" "i686-linux" "x86_64-darwin" "aarch64-linux" ];
+      forAllSystems = f: nixpkgs.lib.genAttrs systems (system: f system);
+      # Memoize nixpkgs for different platforms for efficiency.
+      nixpkgsFor = forAllSystems (
+        system:
+        import nixpkgs {
+          inherit system;
+          overlays = [ self.overlay ];
+        }
+      );
+    in
     {
       overlay = import ./overlay.nix;
+
+      packages = forAllSystems (system: {
+        inherit (nixpkgsFor.${system}) poetry;
+      }
+      );
+
+      apps = forAllSystems (system: {
+        poetry = {
+          type = "app";
+          program = self.packages."${system}".poetry + "/bin/poetry";
```

Comment:
Why is poetry the default app here instead of `poetry2nix`?

Reply:
I never had to use poetry2nix, more like the nix code (mkPoetryEnv etc). This could be added afterwards